tdigest/include/tdigest_impl.hpp

```diff
@@ -282,68 +275,76 @@ double tdigest<T, A>::weighted_average(double x1, double w1, double x2, double w
 }
 
 template<typename T, typename A>
-void tdigest<T, A>::serialize(std::ostream& os) const {
-  const_cast<tdigest*>(this)->merge_buffered(); // side effect
+void tdigest<T, A>::serialize(std::ostream& os, bool with_buffer) const {
+  if (!with_buffer) const_cast<tdigest*>(this)->compress(); // side effect
   write(os, is_empty() || is_single_value() ? PREAMBLE_LONGS_EMPTY_OR_SINGLE : PREAMBLE_LONGS_MULTIPLE);
```

Comment:
can call the newly added `get_preamble_longs()` here

Reply:
right